src/libmongoc/doc/mongoc_handshake.rst

```diff
@@ -0,0 +1,80 @@
+:man_page: mongoc_handshake_data_append
+
+mongoc_handshake_data_append()
+=========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+   bool
+   mongoc_handshake_data_append (const char *driver_name,
+                                 const char *driver_version,
+                                 const char *platform);
+
+Appends the given strings to the handshake data for the underlying C Driver.
+
+Description
+-----------
+
+This function is intended for use by drivers which wrap the C Driver.
+Calling this function will store the given strings as handshake data about
+the system and driver by appending them to the handshake data for the
+underlying C Driver. These values, along with other handshake data collected
+during mongoc_init will be sent to the server as part of the initial
+connection handshake in the "client" document. This function must not be
+called more than once, or after a handshake has been initiated.
```

Comment:
I suggest clarifying the handshake is initiated when server monitoring begins. The [connection pooling](http://mongoc.org/libmongoc/current/connection-pooling.html) page gives additional background into the distinction of singled / pooled monitoring. Feel free to reword the suggestion.

```suggestion
called more than once, or after server monitoring begins. For a single-threaded :symbol:`mongoc_client_t`, server monitoring begins on the first operation requiring a server. For a :symbol:`mongoc_client_pool_t`, server monitoring begins on the first call to `:symbol:`mongoc_client_pool_pop`.
```